lib/web/apiserver.go

```diff
@@ -2355,6 +2360,132 @@ func createIdentityContext(login string, sessionCtx *SessionContext) (srv.Identi
 	}, nil
 }
 
+type UILock struct {
+	Name      string           `json:"name"`
+	Message   string           `json:"message"`
+	Expires   string           `json:"expires"`
+	CreatedAt string           `json:"createdAt"`
+	CreatedBy string           `json:"createdBy"`
+	Targets   types.LockTarget `json:"targets"`
+}
+
+func (h *Handler) getClusterLocks(
+	w http.ResponseWriter,
+	r *http.Request,
+	p httprouter.Params,
+	sessionCtx *SessionContext,
+	site reversetunnel.RemoteSite) (interface{}, error) {
+	ctx := r.Context()
+	clt, err := sessionCtx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	locks, err := clt.GetLocks(ctx, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// Lock data structure is reformatted to save doing it on the client as the
+	// Table component doesn't support nested complex objects. And fails to properly
+	// sort or filter results.
+	lockList := make([]UILock, 0, len(locks))
+	for i := 0; i < len(locks); i++ {
+		expires := ""
+		createdAt := ""
```

Comment:
```suggestion
		var expires, createdAt string
```

Up to you, but I find this a little cleaner.

Reply:
I agree, updated.